addons/account/demo/account_demo.py

```diff
@@ -0,0 +1,352 @@
+import logging
+import time
+from datetime import timedelta
+from dateutil.relativedelta import relativedelta
+import base64
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.misc import file_open
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountChartTemplate(models.Model):
+    _inherit = "account.chart.template"
+
+    @api.model
+    def _get_demo_data(self):
+        yield self._get_demo_data_move()
+        yield self._get_demo_data_statement()
+        yield self._get_demo_data_reconcile_model()
+        yield self._get_demo_data_attachment()
+        yield self._get_demo_data_mail_message()
+        yield self._get_demo_data_mail_activity()
+
+    @api.model
+    def _get_demo_data_move(self):
+        cid = self.env.company.id
+        ref = self.env.ref
+        return ('account.move', [
+            (f'{cid}_demo_invoice_1', {
+                'move_type': 'out_invoice',
+                'partner_id': ref('base.res_partner_12').id,
+                'invoice_user_id': ref('base.user_demo').id,
+                'invoice_payment_term_id': ref('account.account_payment_term_end_following_month').id,
+                'invoice_date': time.strftime('%Y-%m')+'-01',
+                'invoice_line_ids': [
+                    (0, 0, {'product_id': ref('product.consu_delivery_02').id, 'quantity': 5}),
+                    (0, 0, {'product_id': ref('product.consu_delivery_03').id, 'quantity': 5}),
+                ],
+            }),
+            (f'{cid}_demo_invoice_2', {
+                'move_type': 'out_invoice',
+                'partner_id': ref('base.res_partner_2').id,
+                'invoice_user_id': False,
+                'invoice_date': time.strftime('%Y-%m')+'-08',
+                'invoice_line_ids': [
+                    (0, 0, {'product_id': ref('product.consu_delivery_03').id, 'quantity': 5}),
+                    (0, 0, {'product_id': ref('product.consu_delivery_01').id, 'quantity': 20}),
+                ],
+            }),
+            (f'{cid}_demo_invoice_3', {
+                'move_type': 'out_invoice',
+                'partner_id': ref('base.res_partner_2').id,
+                'invoice_user_id': False,
+                'invoice_date': time.strftime('%Y-%m')+'-08',
+                'invoice_line_ids': [
+                    (0, 0, {'product_id': ref('product.consu_delivery_01').id, 'quantity': 5}),
+                    (0, 0, {'product_id': ref('product.consu_delivery_03').id, 'quantity': 5}),
+                ],
+            }),
+            (f'{cid}_demo_invoice_followup', {
+                'move_type': 'out_invoice',
+                'partner_id': ref('base.res_partner_2').id,
+                'invoice_user_id': ref('base.user_demo').id,
+                'invoice_payment_term_id': ref('account.account_payment_term_immediate').id,
+                'invoice_date': (fields.Date.today() + timedelta(days=-15)).strftime('%Y-%m-%d'),
```

Comment:
No chance of issues if this crosses a year boundary?

Reply:
I kept the demo data untouched.